storage/replica.go

```diff
@@ -442,6 +443,15 @@ func (r *Replica) setDescWithoutProcessUpdate(desc *proto.RangeDescriptor) {
 	atomic.StorePointer(&r.desc, unsafe.Pointer(desc))
 }
 
+// getTruncatedState atomically returns the range's truncated state.
+func (r *Replica) getTruncatedState() *proto.RaftTruncatedState {
```

Comment:
Because `Replica` already has a `raftTruncatedState()` method, perhaps this should explicitly be named `cachedTruncatedState()` or something to that effect.